### PR TITLE
Embed CNI configmap in prow CI templates

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

**What this PR does / why we need it**: Apply changes from https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1287/files#diff-2d2faf7aecdc4844f880039bc558bf1054f7807269a2c0dad794276e90dee1e4R16 to the other CI templates so that CNI is applied automatically when deploying these from ci-entrypoint or others.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Embed CNI configmap in prow CI templates
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
@chewong I didn't touch test/dev/cluster-template-custom-builds.yaml, I'll let you add it if needed in #1264

File: templates/test/ci/prow-machine-pool-windows/kustomization.yaml [hunk 1]
```diff
@@ -3,8 +3,18 @@ kind: Kustomization
...
 resources:
   - ../../../flavors/machinepool-windows
-  - ../prow/cni-resource-set-windows.yaml
+  - ../prow-windows/cni-resource-set-windows.yaml
 patchesStrategicMerge:
   - ../patches/tags.yaml
   - ../patches/cluster-cni.yaml
   - ../patches/controller-manager.yaml
+configMapGenerator:
+  - name: cni-${CLUSTER_NAME}-calico
```
Comment (chewong) [inline templates/test/ci/prow-machine-pool-windows/kustomization.yaml:12]:
```suggestion
  - name: cni-${CLUSTER_NAME}-flannel
```

File: templates/test/ci/prow-ipv6/kustomization.yaml [hunk 2]
```diff
@@ -3,8 +3,18 @@ kind: Kustomization
...
 resources:
   - ../../../flavors/ipv6
-  - cni-resource-set-ipv6.yaml
+  - ../prow/cni-resource-set.yaml
 patchesStrategicMerge:
   - ../patches/tags.yaml
   - ../patches/controller-manager.yaml
   - ../patches/cluster-cni.yaml
+configMapGenerator:
+  - name: cni-${CLUSTER_NAME}-calico
```
Comment (chewong) [inline templates/test/ci/prow-ipv6/kustomization.yaml:12]:
```suggestion
  - name: cni-${CLUSTER_NAME}-calico-ipv6
```

Comment (CecileRobertMichon) [reply templates/test/ci/prow-ipv6/kustomization.yaml:12]: it's using ../prow/cni-resource-set.yaml so the configMap name is actually the same, it's just the content that's changing

File: scripts/ci-conformance.sh [hunk 3]
```diff
@@ -50,7 +50,7 @@ get_random_region() {
     echo "${REGIONS[${RANDOM} % ${#REGIONS[@]}]}"
 }
 
-export LOCAL_ONLY=${LOCAL_ONLY:-"false"}
+export LOCAL_ONLY=${LOCAL_ONLY:-"true"}
```
Comment (CecileRobertMichon) [inline scripts/ci-conformance.sh:53]: @chewong you might need to update any jobs that need to push conformance artifacts to an outside registry to have LOCAL_ONLY set to true.

Comment (CecileRobertMichon) [reply scripts/ci-conformance.sh:53]: https://github.com/kubernetes/test-infra/pull/21953

**Review by chewong — COMMENTED** [review]
/lgtm

**Comment by nader-ziada** [comment]
/lgtm 

approve pending the test-infra PR merge

**Comment by nader-ziada** [comment]
/approve

test-infra pr merged
/hold cancel

**Comment by CecileRobertMichon** [comment]
/retest

flake from #1328

**Comment by CecileRobertMichon** [comment]
/retest

:(

**Comment by CecileRobertMichon** [comment]
/retest 

@jsturtevant getting hit by the windows deploy issue repetitively here

**Comment by jsturtevant** [comment]
> getting hit by the windows deploy issue repetitively here

Seems like this got worse today. I will try to repo it

**Comment by CecileRobertMichon** [comment]
I'm going to override the windows job, I've verified that all clusters have built successfully and the windows deployment issue is unrelated to this change. This PR fixes the conformance jobs and unblocks #1330.

/override pull-cluster-api-provider-azure-e2e-windows

**Pull request merged**